server/src/main/java/org/opensearch/wlm/QueryGroupTask.java

```diff
@@ -47,6 +51,20 @@ public QueryGroupTask(
         super(id, type, action, description, parentTaskId, headers, cancelAfterTimeInterval);
```

Comment:
Do we not need to update this constructor? Else `nanoTimeSupplier` will be null?

Reply:
Good Catch!, I will make the member variable supplier final to ensure we initialise it at the creation time